src/rules/no-if.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const {
+  getDocsUrl,
+  testCaseNames,
+  getNodeName,
+  isTestCase,
+} = require('./util');
+
+const isTestArrowFunction = node =>
+  node &&
+  node.type === 'ArrowFunctionExpression' &&
+  node.parent.type === 'CallExpression' &&
+  testCaseNames[getNodeName(node.parent.callee)];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Disallow conditional logic',
+      category: 'Best Practices',
+      recommended: false,
+      uri: getDocsUrl('jest/no-if'),
+    },
+    messages: {
+      noIf: [
+        'Tests should not contain if statements.',
+        'This is usually an indication that you',
+        'are attempting to test too much at once',
+        'or not testing what you intend to.',
+        'Consider breaking the if statement out',
+        'into a separate test to resolve this error.',
+      ].join(' '),
+      noConditional: [
+        'Tests should not contain conditional statements.',
+        'This is usually an indication that you',
+        'are attempting to test too much at once',
+        'or not testing what you intend to.',
+        'Consider writing a separate test for',
+        'each fork in the conditional statement.',
+        'If your conditionals are required to',
+        'satisfy the typescript type checker, consider',
+        'using a non-null assertion operator (!) instead.',
+      ].join(' '),
+    },
+  },
+
+  create(context) {
+    const stack = [];
+
+    function validate(node) {
+      if (!stack[stack.length - 1]) {
```

Comment:
Could you refactor this a bit? I had to read it once or twice and look at what's pushed to `stack` to understand this. Maaaaybe something like


```js
const lastElementInStack = stack[stack.length - 1];

if (lastElementInStack === false) {
  return;
}
```

The reason I added the explicit `=== false` here was because I initially read it as you just check if the element was there, which is not what happens (I _think_) 🙂

Reply:
@SimenB I updated the PR with this change. The `!` (versus explicit `=== false`) also caught cases when the stack was empty so I amended with a more explicit check for that case too.

Reply:
thanks, easier to follow now!